src/Chapter3.hs

```diff
@@ -374,6 +382,40 @@ after the fight. The battle has the following possible outcomes:
 
 -}
 
+data Fighter' = 
+    KnightF  { fighterHealth :: Int
+             , fighterAttack :: Int
+             , fighterGold :: Int
+             }
+
+  | MonsterF { fighterHealth :: Int
+             , fighterAttack :: Int
+             , fighterGold :: Int
+             }
+  deriving (Show)
```

Comment:
Nice usage of the sum type!

The only issue with that (the reusage of the same data type for Monster and Knight) in here is that you can easily mix the constructors when calling the function, and the compiler won't tell you anything about that:

```haskell
> fight knight monster
> fight monster knight
```

Both of the calls are absolutely valid, but the results are going to be different!

Here, we had in mind to make two types for each Monster and Knight.